### PR TITLE
fix: set `useCdn: false` automatically when `perspective: 'previewDrafts'`

### PR DESCRIPTION
**Pull request by stipsan**

### CODE REVIEW

File: src/data/dataMethods.ts [hunk 1]
```diff
@@ -323,6 +323,10 @@ export function _requestObservable<R>(
     if (typeof perspective === 'string' && perspective !== 'raw') {
       validateApiPerspective(perspective)
       options.query = {perspective, ...options.query}
+      // If the perspective is set to `previewDrafts` we can't use the CDN, the API will throw
+      if (perspective === 'previewDrafts') {
+        useCdn = false
+      }
```
Comment (stipsan) [inline src/data/dataMethods.ts]: Should we warn when this happens?

E.g. something like
```suggestion
      if (useCdn && perspective === 'previewDrafts') {
        useCdn = false
        console.warn(`perspective: 'previewDrafts' isn't supported with useCdn: true`)
      }
```

Comment (rexxars) [reply src/data/dataMethods.ts]: I think we should. We have a warning printer which ensures we only print once - should probably use that?
https://github.com/sanity-io/client/blob/main/src/warnings.ts#L5

File: src/warnings.ts [hunk 2]
```diff
@@ -12,6 +12,11 @@ export const printCdnWarning = createWarningPrinter([
   `\`useCdn: false\` to use the Live API. Note: You may incur higher costs using the live API.`,
 ])
 
+export const printCdnPrevieDraftsWarning = createWarningPrinter([
+  `The Sanity client is configured with the \`perspective\` set to \`previewDrafts\`, which doesn't support the API-CDN.`,
+  `The Live API will be used instead. Set \`useCdn: false\` in your configuration to hide this warning.`,
```
Comment (stipsan) [inline src/warnings.ts:17]: @rexxars does this look ok?

File: src/warnings.ts [hunk 3]
```diff
@@ -12,6 +12,11 @@ export const printCdnWarning = createWarningPrinter([
   `\`useCdn: false\` to use the Live API. Note: You may incur higher costs using the live API.`,
 ])
 
+export const printCdnPrevieDraftsWarning = createWarningPrinter([
```
Comment (rexxars) [inline src/warnings.ts]:
```suggestion
export const printCdnPreviewDraftsWarning = createWarningPrinter([
```

File: src/data/dataMethods.ts [hunk 4]
```diff
@@ -323,6 +324,11 @@ export function _requestObservable<R>(
     if (typeof perspective === 'string' && perspective !== 'raw') {
       validateApiPerspective(perspective)
       options.query = {perspective, ...options.query}
+      // If the perspective is set to `previewDrafts` we can't use the CDN, the API will throw
+      if (perspective === 'previewDrafts') {
+        useCdn = false
+        printCdnPrevieDraftsWarning()
```
Comment (rexxars) [inline src/data/dataMethods.ts]:
```suggestion
        printCdnPreviewDraftsWarning()
```

File: src/warnings.ts (hunk 2 above)
Comment (rexxars) [reply src/warnings.ts:17]: Message looks okay to me - suggested a typo fix.

**Review by rexxars — APPROVED** [review]

**Pull request merged**